### PR TITLE
Rename "hit effect" for hazards to "target effect".

### PR DESCRIPTION
**Pull request by petervdmeer**
**Feature:** This PR replaces the "hit effect" keyword for hazards by "target effect" as discussed in #5751

## Feature Details
This is just the rename to differentiate between "hit effect" (effect at point of impact, for projectile weapons) and "target effect" (as effects over a ship shape, for hazards) and some changes to make the hazard-weapon-terminology more in line with the projectile-weapon-terminilogy.
There are a number of other improvements being discussed in #5751, but the discussion was just started so I want to give a bit more time to discuss before creating other improvements.

## UI Screenshots
N/A (only the keyword changed)

## Usage Examples
See `data/hazards.txt` for the replaced keyword.

## Testing Done
Flown to Sagittarius A* and got my ship destroyed there in a cloud of nano particles.

## Performance Impact
None expected.

### CODE REVIEW

**Comment by Zitchas** [comment]
Sure, I don't see a problem with this. Keeps things a bit more separate.

**Comment by petervdmeer** [comment]
> Sure, I don't see a problem with this. Keeps things a bit more separate.

The actual goal is to make things more integrated; enable the ability to use the hazard keyword on projectile weapons; and enable the ability to use the projectile keyword on hazards. (And that is only possible if the keywords don't overlap.)

I hope that this will also help us to re-use more code between projectile weapon damage and hazard damage. (Thus also reducing a number of separate code-paths to single code paths that handle both.)

File: source/Ship.cpp [hunk 1]
```diff
@@ -3048,16 +3047,16 @@ int Ship::TakeDamage(const Projectile &projectile, bool isBlast)
 
 
 
-// This ship just got hit by the given hazard. Take damage according to what
+// This ship just got hit by an hazard. Take damage according to what
```
Comment (Amazinite) [inline source/Ship.cpp:3050]:
```suggestion
// This ship just got hit by a hazard. Take damage according to what
```

File: source/Ship.h [hunk 2]
```diff
@@ -323,9 +322,9 @@ class Ship : public Body, public std::enable_shared_from_this<Ship> {
 	// not necessarily its primary target.
 	// Blast damage is dependent on the distance to the damage source.
 	int TakeDamage(const Projectile &projectile, bool isBlast = false);
-	// This ship just got hit by the given hazard. Take damage according to what
+	// This ship just got hit by an hazard. Take damage according to what
```
Comment (Amazinite) [inline source/Ship.h:325]:
```suggestion
	// This ship just got hit by a hazard. Take damage according to what
```

**Review by Amazinite — COMMENTED** [review]
So I'm assuming a future change might allow normal weaponry to also make use of this target effect.

**Comment by petervdmeer** [comment]
> So I'm assuming a future change might allow normal weaponry to also make use of this target effect.

Yes, that is the next step.

**Pull request merged**